sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchemaSuite.scala

```diff
@@ -73,6 +73,12 @@ class CSVInferSchemaSuite extends SparkFunSuite {
     assert(CSVInferSchema.inferField(LongType, "2015-08 14:49:00", options) == StringType)
   }
 
+  test("Boolean field types are inferred correctly from custom value") {
+    val options = new CSVOptions(Map("trueValue" -> "yes"))
+    assert(CSVInferSchema.inferField(BooleanType, "yES", options) == BooleanType)
```

Comment:
No assert in this suite uses `===`. Should I make this consistent and replace all occurrences of `==` in that suite with `===` or just the one in the lines I've changed?